server/src/main/java/org/apache/iotdb/db/mpp/transformation/dag/column/unary/scalar/CastFunctionColumnTransformer.java

```diff
@@ -0,0 +1,227 @@
+            String.format("Unsupported target dataType: %s", returnType.getTypeEnum()));
+    }
+  }
+
+  private void cast(ColumnBuilder columnBuilder, float value) {
+    switch (returnType.getTypeEnum()) {
+      case INT32:
+        returnType.writeInt(columnBuilder, (int) value);
+        break;
+      case INT64:
+        returnType.writeLong(columnBuilder, (long) value);
+        break;
+      case FLOAT:
+        returnType.writeFloat(columnBuilder, value);
+        break;
+      case DOUBLE:
+        returnType.writeDouble(columnBuilder, value);
+        break;
+      case BOOLEAN:
+        returnType.writeBoolean(columnBuilder, value != 0.0f);
+        break;
+      case BINARY:
+        returnType.writeBinary(columnBuilder, Binary.valueOf(String.valueOf(value)));
+        break;
+      default:
+        throw new UnsupportedOperationException(
+            String.format("Unsupported target dataType: %s", returnType.getTypeEnum()));
+    }
+  }
+
+  private void cast(ColumnBuilder columnBuilder, double value) {
+    switch (returnType.getTypeEnum()) {
+      case INT32:
+        returnType.writeInt(columnBuilder, (int) value);
+        break;
+      case INT64:
+        returnType.writeLong(columnBuilder, (long) value);
+        break;
+      case FLOAT:
+        returnType.writeFloat(columnBuilder, (float) value);
+        break;
+      case DOUBLE:
+        returnType.writeDouble(columnBuilder, value);
+        break;
+      case BOOLEAN:
+        returnType.writeBoolean(columnBuilder, value != 0.0);
+        break;
+      case BINARY:
+        returnType.writeBinary(columnBuilder, Binary.valueOf(String.valueOf(value)));
+        break;
+      default:
+        throw new UnsupportedOperationException(
+            String.format("Unsupported target dataType: %s", returnType.getTypeEnum()));
+    }
+  }
+
+  private void cast(ColumnBuilder columnBuilder, boolean value) {
+    switch (returnType.getTypeEnum()) {
+      case INT32:
+        returnType.writeInt(columnBuilder, value ? 1 : 0);
+        break;
+      case INT64:
+        returnType.writeLong(columnBuilder, value ? 1L : 0);
+        break;
+      case FLOAT:
+        returnType.writeFloat(columnBuilder, value ? 1.0f : 0);
+        break;
+      case DOUBLE:
+        returnType.writeDouble(columnBuilder, value ? 1.0 : 0);
+        break;
+      case BOOLEAN:
+        returnType.writeBoolean(columnBuilder, value);
+        break;
+      case BINARY:
+        returnType.writeBinary(columnBuilder, Binary.valueOf(String.valueOf(value)));
+        break;
+      default:
+        throw new UnsupportedOperationException(
+            String.format("Unsupported target dataType: %s", returnType.getTypeEnum()));
+    }
+  }
+
+  private void cast(ColumnBuilder columnBuilder, Binary value) {
+    String stringValue = value.getStringValue();
+    switch (returnType.getTypeEnum()) {
+      case INT32:
+        returnType.writeInt(columnBuilder, (int) Double.parseDouble(stringValue));
```

Comment:
I found that if we directly use Integer.parseInt for string value `1.1`, it will throw exception, so I wonder that how pg or mysql(standard sql) deal with this case?